libsandbox/libsandbox.c

```diff
@@ -349,14 +349,15 @@ char *egetcwd(char *buf, size_t size)
 
 	/* If tracing a child, our cwd may not be the same as the child's */
 	if (trace_pid) {
-		char proc[20];
-		sprintf(proc, "/proc/%i/cwd", trace_pid);
-		ssize_t ret = readlink(proc, buf, size);
-		if (ret == -1) {
+		char proc[21] = { 0 };
+		snprintf(proc, sizeof(proc), "/proc/%i/cwd", trace_pid);
+		proc[sizeof(proc) - 1] = '\0';
```

Comment:
snprintf always writes a trailing `'\0'` when the buffer size is greater than zero, so this is unnecessary.

Reply:
Ah good point.